model_api/python/model_api/models/visual_prompting.py

```diff
@@ -157,6 +158,7 @@ def __init__(
              Once the features are passed, one can skip learn() method, and start predicting masks right away. Defaults to None.
             threshold (float, optional): Threshold to match vs reference features on infer(). Greater value means a
             stricter matching. Defaults to 0.65.
+            masks_extra_refinement (bool, optional): Flag controlling additional refinement stage on inference.
```

Comment:
I suggest passing this as an argument to `infer()` rather than in the constructor, as it seems only relevant there. Also, consider renaming it to `apply_masks_refinement` for clarity. 

It would be helpful to add more context to the docstring, briefly describing the effect of enabling or disabling this option.

Reply:
Thanks for suggestion, I've updated the implementation